로그인 구현

Hosick opened the pull request:
- 로그인 / 로그아웃 구현
- 세션을 이용한 로그인 유지 구현
- 세션 timeout 변경(24시간)
- 로그인 유저 정보 가져오기 구현
- 핸들러 인터셉터와 어노테이션을 이용한 로그인 검사 구현

Hosick committed:
etc: applicaition.yml 수정

-오타 수정
-주석 수정

[#1] 프로젝트 설정

Hosick committed:
[#1] application.yml 분리

Hosick committed:
[#1]chore: yml 분리

- 모든 profile에서 사용되는 전역설정 분리

Hosick committed:
[#1]chore: 리뷰 반영

-logging 옵션 분리 적용
-production profile 용 yml 생성

Hosick committed:
[#1]chore: properties 설정 불필요한 값 제거

Hosick committed:
[#8] 회원가입 시 SMS 인증 구현

-redis 기본 설정
-인증번호 발송 시 redis에 휴대폰 번호와 인증번호를 key:value 형식으로 저장
-메모리에 저장된 인증번호 값에 제한시간  부여

Hosick committed:
[#6]feat: 로그인 구현

-로그인 / 로그아웃 구현
-세션을 이용한 로그인 유지 구현
-세션 timeout 변경(24시간)
-로그인 유저 정보 가져오기 구현
-핸들러 인터셉터와 어노테이션을 이용한 로그인 검사 구현

Hosick commented on src/main/java/com/flab/shoeauction/config/WebConfig.java:
```diff
@@ -0,0 +1,19 @@
+
+    @Override
+    public void addInterceptors(InterceptorRegistry registry) {
+        registry.addInterceptor(loginInterceptor);
```
어노테이션을 이용한 로그인 검사를 하게되면 모든 핸들러는 타기전에  우선 핸들러 인터셉터를 거치게 되는데 거기서 오는 비용은 신경쓰지 않아도 되는 정도일까요?

f-lab-dev replied:
먼저 호석님은 어떻게 생각하시나요?

Hosick replied:
어느정도 비용이 들거라고 생각되었었는데 막상 찾아보니 관련 주의사항도 찾을 수 없고, 인터셉터에서 로그로 시간을 찍어봐도 미미한 것 같습니다.

f-lab-dev commented on src/main/java/com/flab/shoeauction/Application.java:
```diff
@@ -2,8 +2,10 @@
+import org.springframework.boot.web.servlet.ServletComponentScan;
 import org.springframework.data.jpa.repository.config.EnableJpaAuditing;
 
+@ServletComponentScan
```
이거는 왜 달아주신걸까요?

Hosick replied:
`HttpSessionListener`를 사용해서 session에 타임아웃을 주려고 사용했었습니다.
properties를 사용해서 타임아웃을 주는 방법이 더 좋을 것 같아서 변경했습니다!

f-lab-dev commented on src/main/java/com/flab/shoeauction/interceptor/LoginInterceptor.java:
```diff
@@ -0,0 +1,32 @@
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler) throws Exception {
+        CheckLogin checkLogin = ((HandlerMethod) handler).getMethodAnnotation(CheckLogin.class);
+
+        if (checkLogin != null && sessionService.getLoginUserEmail() == null)
```
`checkLogin != null && sessionService.getLoginUserEmail() == null` 이거를 메소드로 분리해서 이름을 지어 사용하면 가독성이 더 높아질 수 있을 것 같네요~

Hosick replied:
넵 앞으로도 최대한 가독성있게 작성하도록 노력해보겠습니다!

f-lab-dev commented on src/main/java/com/flab/shoeauction/interceptor/LoginInterceptor.java:
```diff
@@ -0,0 +1,32 @@
+
+        if (checkLogin != null && sessionService.getLoginUserEmail() == null)
+            throw new NotLoginException();
+        return true;
```
요런곳에도 위에 개행 하나 넣어주시면 더 깔끔할 것 같아요~

f-lab-dev commented on src/main/java/com/flab/shoeauction/web/UserApiController.java:
```diff
@@ -29,6 +38,28 @@
+    }
+
+    @CheckLogin
+    @GetMapping("/loggedin-user")
```
현재 로그인된 유저의 정보라고 하면 **내 정보** 아닐까요?

Hosick replied:
앗 맞습니다. 사용자 입장에서 네이밍했어야 했는데 저도 모르게 간과한 것 같습니다.
수정하겠습니다!

Hosick committed:
merge branch settings

Hosick committed:
[#6] 리뷰 반영

-sessionListner 삭제 및 properties에 timeout 설정
-interceptor 내 로그인 검사 메소드로 분리
-로그인 정보 가져오기 /my-infos로 변경

Hosick committed:
merge settings branch

Hosick committed:
[#8] 리뷰 반영

-인증번호 prefix 변수명 수정
-redis set, expire 명령 하나로 통합

f-lab-dev commented on src/main/java/com/flab/shoeauction/interceptor/LoginInterceptor.java:
```diff
@@ -21,12 +21,20 @@
+    }
+
+    private boolean isNotLogin(CheckLogin checkLogin) {
+        if (checkLogin == null) return false;
```
여기는 개행해주는게 더 깔끔할 것 같습니다~ 이렇게 한줄로 늘어놓는다고 가독성이 좋아지는건 아니니까요

Hosick replied:
짧은 조건문을 작성해야 할 때 한줄로 만드는 습관이 있는데 일관되지 않고 가독성에도 좋지 않은 습관이라고 생각됩니다.
디테일한 리뷰 감사합니다!

f-lab-dev commented on src/main/resources/application.yml:
```diff
@@ -4,4 +4,6 @@ spring:
\ No newline at end of file
+        format_sql: true
+
+server.servlet.session.timeout : 24h
```
타임아웃이 꽤나 긴 것 같은데 이렇게 길게 설정하신 이유가 있으실까요?

Hosick replied:
앗 이 부분은 제가 평소에 이용하던 사이트들의 `로그인 상태 유지` 기능을 떠올려서 생각없이 작성한 것 같습니다.
차라리 일반적인 로그인의 경우 default timeout인 30분으로 설정되게 하고, 추후에 `로그인 상태 유지` 기능을 이슈로 빼서 따로 구현하도록 하겠습니다!

Hosick committed:
[#6] 리뷰 반영

-session timeout 삭제(추후에 로그인 상태 유지 기능 따로 구현 예정)
-개행 수정

Hosick committed:
Merge branch 'feature/8/hs' into feature/6/hs
